e2e/runner/setup_ton.go

```diff
@@ -55,7 +66,58 @@ func (r *E2ERunner) SetupTON() error {
 	}
 
 	r.TONDeployer = deployer
-	r.TONGateway = gwAccount.ID
+	r.TONGateway = toncontracts.NewGateway(gwAccount.ID)
 
-	return nil
+	return r.ensureTONChainParams(gwAccount)
+}
+
+func (r *E2ERunner) ensureTONChainParams(gw *ton.AccountInit) error {
+	if r.ZetaTxServer == nil {
+		return errors.New("ZetaTxServer is not initialized")
+	}
+
+	creator := r.ZetaTxServer.MustGetAccountAddressFromName(utils.OperationalPolicyName)
+
+	chainID := chains.TONLocalnet.ChainId
+
+	chainParams := &observertypes.ChainParams{
+		ChainId:                     chainID,
+		ConfirmationCount:           1,
+		GasPriceTicker:              5,
+		InboundTicker:               5,
+		OutboundTicker:              5,
+		ZetaTokenContractAddress:    constant.EVMZeroAddress,
+		ConnectorContractAddress:    constant.EVMZeroAddress,
+		Erc20CustodyContractAddress: constant.EVMZeroAddress,
+		OutboundScheduleInterval:    2,
+		OutboundScheduleLookahead:   5,
+		BallotThreshold:             observertypes.DefaultBallotThreshold,
+		MinObserverDelegation:       observertypes.DefaultMinObserverDelegation,
+		IsSupported:                 true,
+		GatewayAddress:              gw.ID.ToRaw(),
+	}
+
+	msg := observertypes.NewMsgUpdateChainParams(creator, chainParams)
+
+	if _, err := r.ZetaTxServer.BroadcastTx(utils.OperationalPolicyName, msg); err != nil {
```

Comment:
other chain params are added during the [genesis](https://github.com/zeta-chain/node/blob/feat%2Fton-observer/x/observer/genesis.go#L13)

Reply:
We've discussed with @lumtis that it's a bad practice. It's also not possible to set TON params during genesis for because the gateway address is always different for each e2e run

Reply:
oh. IC.